src/transport_ofi.c

```diff
@@ -1440,6 +1433,9 @@ int shmem_transport_init(void)
                            shmem_internal_params.OFI_STX_MAX);
         }
         shmem_transport_ofi_stx_max = 1;
+    } else if (shmem_internal_params.OFI_STX_AUTO) {
+        ret = shmem_node_util_init();
```

Comment:
Should node utils be initialized by the transport or as part of the top-level init?  The latter gives cleaner separation between these components.

Reply:
Do you mean move this check for `OFI_STX_AUTO` to `init.c`?  I'd think we'd prefer to keep OFI specific parameters out of the top-level init.  It might be nice to have `USE_ON_NODE_COMMS` capture `OFI_STX_AUTO`, but since it's set at configure time, I'm not sure how to do that.

Reply:
You could add a query function like `shmem_transport_needs_node_utils()`